src/backend/crossterm.rs

```diff
@@ -4,28 +4,147 @@
+            KeyCode::End => Key::End,
+            KeyCode::PageUp => Key::PageUp,
+            KeyCode::PageDown => Key::PageDown,
+            KeyCode::Delete => Key::Del,
+            KeyCode::Insert => Key::Ins,
+            KeyCode::Enter => Key::Enter,
+            KeyCode::Tab => Key::Tab,
+            KeyCode::F(n) => Key::from_f(n),
+            KeyCode::BackTab => Key::Tab, /* not supported */
+            KeyCode::Char(_) => Key::Tab, /* is handled at `Event` level, use tab as default */
+            KeyCode::Null => Key::Tab, /* is handled at `Event` level, use tab as default */
+        }
+    }
+}
+
+impl From<CKeyEvent> for Event {
+    fn from(event: CKeyEvent) -> Self {
+        const CTRL_ALT: KeyModifiers = KeyModifiers::from_bits_truncate(
+            KeyModifiers::CONTROL.bits() | KeyModifiers::ALT.bits(),
+        );
+        const CTRL_SHIFT: KeyModifiers = KeyModifiers::from_bits_truncate(
+            KeyModifiers::CONTROL.bits() | KeyModifiers::SHIFT.bits(),
+        );
+        const ALT_SHIFT: KeyModifiers = KeyModifiers::from_bits_truncate(
+            KeyModifiers::ALT.bits() | KeyModifiers::SHIFT.bits(),
+        );
+
+        match event {
+            // Handle Char + modifier.
+            CKeyEvent {
+                modifiers: KeyModifiers::CONTROL,
+                code: KeyCode::Char('c'),
+            } => Event::Exit,
+            CKeyEvent {
+                modifiers: KeyModifiers::CONTROL,
+                code: KeyCode::Char(c),
+            } => Event::CtrlChar(c),
+            CKeyEvent {
+                modifiers: KeyModifiers::ALT,
+                code: KeyCode::Char(c),
+            } => Event::AltChar(c),
+            CKeyEvent {
+                modifiers: KeyModifiers::SHIFT,
+                code: KeyCode::Char(c),
+            } => Event::Char(c),
+
+            // Handle key + multiple modifiers
+            CKeyEvent {
+                modifiers: CTRL_ALT,
+                code,
+            } => Event::CtrlAlt(Key::from(code)),
+            CKeyEvent {
+                modifiers: CTRL_SHIFT,
+                code,
+            } => Event::CtrlShift(Key::from(code)),
+            CKeyEvent {
+                modifiers: ALT_SHIFT,
+                code,
+            } => Event::AltShift(Key::from(code)),
+
+            // Handle key + single modifier
+            CKeyEvent {
+                modifiers: KeyModifiers::CONTROL,
+                code,
+            } => Event::Ctrl(Key::from(code)),
+            CKeyEvent {
+                modifiers: KeyModifiers::ALT,
+                code,
+            } => Event::Alt(Key::from(code)),
+            CKeyEvent {
+                modifiers: KeyModifiers::SHIFT,
+                code,
+            } => Event::Shift(Key::from(code)),
+
+            CKeyEvent {
+                modifiers: _,
+                code: KeyCode::Char(c),
+            } => Event::Char(c),
+            CKeyEvent { modifiers: _, code } => Event::Key(Key::from(code)),
```

Comment:
maybe
```rust
            CKeyEvent {
                modifiers: _,
                code: KeyCode::BackTab,
            } => Event::Shift(Key::Tab),
```
(this makes shift-tab work the same as with the ncurses backend, only tested on ubuntu)

Reply:
Doesn't this already work? When I press SHIFT + TAB, crossterm sends a Backtap key, this BACK_TAP key is translated to TAB [over here](https://github.com/TimonPost/cursive/blob/crossterm_update/src/backend/crossterm.rs#L75). And will be [parsed here.](https://github.com/TimonPost/cursive/blob/crossterm_update/src/backend/crossterm.rs#L139)

Reply:
it didn't work for me, shift-tab was interpreted as forward tab (both in the `keycode` example and in `radio`)

Reply:
Fixed